pxl_scripts/sotw/dns_queries_filtered/data.pxl

```diff
@@ -0,0 +1,44 @@
+ # Copyright (c) Pixie Labs, Inc.
+# Licensed under the Apache License, Version 2.0 (the "License")
+
+''' DNS Queries filtered by query name. 
+Lists all DNS queries filtered by a specific query name. 
+'''
+
+import px
+
+def dns_requests(start: str, query_name_filter: str):
+
+    # The dns_events table pairs DNS requests with their responses.
+	df = px.DataFrame(table='dns_events', start_time=start)
+
+	# Add context.
+	df.pod = df.ctx['pod']
+
+	# Convert DNS IP to string.
+	df.dns_server = px.Service(px.nslookup(df.remote_addr))
+
+	# Parse the DNS respoonse to determine if it was successfully resolved.
+	df.resp_body = px.pluck(df.resp_body, 'answers')
+	df.request_resolved = px.contains(df.resp_body, 'name')
+
+	# Parse the DNS request for query name.
+	df.req_body = px.pluck(df.req_body, 'queries')
+	df.idx1 = px.find(df.req_body, '\"name\":')
```

Comment:
similar, note here that this logic will get cleaner when we add support for object types